src/lib.rs

```diff
@@ -204,29 +218,37 @@ impl<'a> VisitMut for IntLiteralVisitor<'a> {
 impl<'a> VisitMut for NumericLiteralVisitor<'a> {
     fn visit_expr_mut(&mut self, expr: &mut Expr) {
         if let Expr::Lit(lit_expr) = expr {
-            match lit_expr.lit {
-                // TODO: Currently we cannot correctly treat integers that don't fit in 64
-                // bits. For this we'd have to deal with verbatim literals and manually
-                // parse the string
-                Lit::Int(_) => {
-                    let mut visitor = IntLiteralVisitor {
-                        parameters: self.parameters,
-                        placeholder: self.placeholder,
-                        replacement: self.int_replacement,
-                    };
-                    visitor.visit_expr_mut(expr);
-                    return;
-                }
-                Lit::Float(_) => {
-                    let mut visitor = FloatLiteralVisitor {
-                        parameters: self.parameters,
-                        placeholder: self.placeholder,
-                        replacement: self.float_replacement,
-                    };
-                    visitor.visit_expr_mut(expr);
-                    return;
-                }
-                _ => {}
+            // TODO: Currently we cannot correctly treat integers that don't fit in 64
+            //  bits. For this we'd have to deal with verbatim literals and manually
+            //  parse the string
+
+            let is_float = match &lit_expr.lit {
+                // Literals like `20f64` are parsed as `LitInt`s
+                Lit::Int(int_lit) => matches!(int_lit.suffix(), "f32" | "f64"),
+                Lit::Float(_) => true,
+                _ => false,
+            };
+
+            if is_float {
+                let mut visitor = FloatLiteralVisitor {
+                    parameters: self.parameters,
+                    placeholder: self.placeholder,
+                    replacement: self.float_replacement,
+                };
+                visitor.visit_expr_mut(expr);
+                return;
+            }
+
+            let is_integer = matches!(lit_expr.lit, Lit::Int(_));
```

Comment:
I don't mind merging as is, but couldn't this be written a bit more idiomatically as a single `match` statement? Hmm, but maybe you can't do something like `Lit::Float(_) | Lit::Int(int_lit) if matches!(int_lit.suffix(), ...)`.

Mostly just curious :) As I said, I'm fine with the code as is.

Reply:
Right, turns out this is also possible. One could also remove this here and always forward all literals to the float and int visitor such that this logic is not duplicated between the visitors? The third alternative would be to use a separate functions like `literal_represents_float() -> bool` and `literal_represents_int() -> bool` which get used by all visitors?